### PR TITLE
[fix] Emitting events from different coroutines

### PR DESCRIPTION
**Pull request by ohassine**
This PR is pair programmed one between @vitorhugods and I.

It aims to improve the way we are emiting  events received from two different sources.

### CODE REVIEW

**Review by vitorhugods — APPROVED** [review]

**Review by vitorhugods — CHANGES_REQUESTED** [review]
As discussed, waiting proposed changes regarding how we deal with closing the flow.

File: app/src/main/kotlin/com/wire/android/core/events/datasource/EventDataSource.kt [hunk 1]
```diff
@@ -7,48 +7,41 @@ import com.wire.android.core.events.datasource.remote.NotificationRemoteDataSour
...
+        }
+
+        val pendingEventsFlow = currentClientIdFlow.mapNotNull { clientId ->
+            val notificationId = lastNotificationId(clientId).fold({ null }, { it })
+            notificationId?.let { clientId to it }
+        }.flatMapMerge {
+            notificationRemoteDataSource.notificationsFlow(it.first, it.second)
+        }.flatMapMerge { it.asFlow() }
+
+        val liveEventsFlow = currentClientIdFlow.flatMapMerge { clientId ->
```
Comment (mejdoo) [inline app/src/main/kotlin/com/wire/android/core/events/datasource/EventDataSource.kt:38]: Out of curiosity, what's the difference between `liveEvents` and `pendingEvents`?

**Review by mejdoo — APPROVED** [review]
Sounds good 👍 I left some minor questions.

File: app/src/main/kotlin/com/wire/android/core/events/datasource/remote/NotificationRemoteDataSource.kt [hunk 2]
```diff
@@ -32,10 +35,11 @@ class NotificationRemoteDataSource(
         webSocketServiceForClient(clientId)
             .observeWebSocketEvent().filter {
                 it is WebSocket.Event.OnConnectionOpened<*>
-            }.map { allNotifications(clientId, notificationId) }
+            }.take(1).map { allNotifications(clientId, notificationId) }
```
[the remark this answers is not in the document]
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/core/events/datasource/remote/NotificationRemoteDataSource.kt:38]: And we added `take()` to cancel the flow after being consumed

**Pull request merged**